src/sentry/models/outbox.py

```diff
@@ -678,11 +686,14 @@ def drain_shard(
                 if _test_processing_barrier:
                     _test_processing_barrier.wait()
 
-                shard_row.process(is_synchronous_flush=not flush_all)
+                processed = shard_row.process(is_synchronous_flush=not flush_all)
```

Comment:
Won't the query in `self.process_shard` still result in the same index scan failure we've been seeing? It looks like the `select_for_update` precedes this which I believe is the point of failure 🤔

Reply:
That's ok because _it already has the lock_ in the current session.  It is re-entrant.

Reply:
The timeout is related to other concurrent processing of the same shard.